### PR TITLE
v3.0, dev: add doc for recover table

### PR DESCRIPTION
**Pull request by crazycs520**
### What is changed, added or deleted? 

Add doc for tidb recover table.

### Checklist 

- [ ] Add a new file to `TOC.md`
- [ ] No Tab spaces anywhere 
- [ ] Leave a blank line both before and after a code block
- [ ] Keep the first level heading consistent with `title` in metadata
- [ ] Use *four* spaces for each level of indentation except that in `TOC.md`

### CODE REVIEW

File: v3.0/reference/sql/statements/recover-table.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+{{< copyable "sql" >}}
+
+```sql
+DROP TABLE t;
+
+RECOVER TABLE t;
+```
+
+根据表名恢复被删除的表，会找到最近历史 DDL JOB 中的第一个是 DROP TABLE 类型的 DDL 且 DROP TABLE 的表名等于 Recover Table 语句中指定的表名的表进行恢复。
```
Comment (zimulala) [inline v3.0/reference/sql/statements/recover-table.md]: `DROP TABLE` 和 `Recover Table` 的格式是否统一一下？

File: v3.0/reference/sql/statements/recover-table.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+## 原理
+
+TiDB 在删除表时，实际上只删除了表的元信息，并将需要删除的表数据（行数据和索引数据）写一条数据到 `mysql.gc_delete_range` 表。TiDB 后台的 GC Worker 会定期从 `mysql.gc_delete_range` 表中取出超过 gc life time 相关范围的 key 进行删除。
+
+所以，Recover Table 只需要在 GC Worker 还没删除表数据前，恢复表的元信息并删除 `mysql.gc_delete_range` 表中相应的行记录就可以了。恢复表的元信息可以用 TiDB 的快照读实现。具体的快照读内容可以参考 [读取历史数据](../../../how-to/get-started/read-historical-data.md) 文档。
+
+恢复表的元信息是通过快照读获取表的元信息后，再走一次类似于 CREATE TABLE 的建表流程，所以 Recover Table 实际上也是一种 DDL。
+
+如果删除表后并过了 gc life time，就不能再用 Recover Table 来恢复被删除的表了。
```
Comment (zimulala) [inline v3.0/reference/sql/statements/recover-table.md]: 这是把这个另外加一个小标题下面，比如注意事项，阅读的时候比较明显。
另外是否说明一下目前不支持此功能的情况

**Comment by crazycs520** [comment]
@lilin90 PTAL

**Comment by yikeke** [comment]
**Important Update**: We have added a markdownlint static check in the CI check to improve the quality of our documentation, so your unclosed PR might fail the CI check now.

Click `details` beside the `ci/circleci: lint — Your tests failed on CircleCI` message, and you can check the details for the failure and fix all the issues in your PR. It is recommended to install "markdownlint" extension in your VS Code editor, so it will check all potential issues that fail the CI check. 

@crazycs520 If you have any problem with troubleshooting, please consult @yikeke for help. Thanks for your cooperation~

Ref: https://github.com/pingcap/docs-cn/pull/1494

File: dev/reference/sql/statements/recover-table.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    {{< copyable "sql" >}}
+
+    ```sql
+    DROP TABLE t;
+    ```
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    ADMIN SHOW DDL JOBS 1; -- 查找删除表 t 时的 DDL JOB ID，这里是 53。
```
Comment (yikeke) [inline dev/reference/sql/statements/recover-table.md]:
```suggestion
    ADMIN SHOW DDL JOBS 1;
```
    查找删除表 t 时的 DDL JOB ID，这里是 53。

File: dev/reference/sql/statements/recover-table.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    ```
+
```
Comment (yikeke) [inline dev/reference/sql/statements/recover-table.md]:
```suggestion

查找删除表 t 时的 DDL JOB ID，这里是 53。

```

File: dev/reference/sql/statements/recover-table.md (hunk 3 above)
Comment (yikeke) [reply dev/reference/sql/statements/recover-table.md]: It is not recommended to add code comments in the copyable code block. You can write it above or below the block instead.

File: v3.0/reference/sql/statements/recover-table.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    {{< copyable "sql" >}}
+
+    ```sql
+    DROP TABLE t;
+    ```
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    ADMIN SHOW DDL JOBS 1; -- 查找删除表 t 时的 DDL JOB ID，这里是 53。
```
Comment (yikeke) [inline v3.0/reference/sql/statements/recover-table.md]:
```suggestion
    ADMIN SHOW DDL JOBS 1; -- 查找删除表 t 时的 DDL JOB ID，这里是 53。
```
ditto

File: dev/reference/sql/statements/recover-table.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+查找删除表 t 时的 DDL JOB ID，这里是 53。
```
Comment (yikeke) [inline dev/reference/sql/statements/recover-table.md]:
```suggestion
    查找删除表 t 时的 DDL JOB ID，这里是 53。
```

File: dev/reference/sql/statements/recover-table.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    +--------+---------+------------+------------+--------------+-----------+----------+-----------+-----------------------------------+--------+
+    ```
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    RECOVER TABLE BY JOB 53;
+    ```
+
+    根据删除表时的 DDL JOB ID 恢复被删除的表，会直接用 DDL JOB ID 找到被删除表进行恢复。如果指定的 DDL JOB ID 的 DDL JOB 不是 DROP TABLE 类型，会报错。
```
Comment (winkyao) [inline dev/reference/sql/statements/recover-table.md]: 这里多加一个解释，为什么除了 recover table name，还需要这个 job id 的语句

Comment (crazycs520) [reply dev/reference/sql/statements/recover-table.md]: done.

**Pull request merged**